Source/Core/DiscIO/FileSystemGCWii.cpp

```diff
@@ -19,83 +21,309 @@
 
 namespace DiscIO
 {
+// Set everything manually
+CFileInfoGCWii::CFileInfoGCWii(const u8* fst, bool wii, u32 fst_size, u32 index, u32 total_file_infos)
+	: m_fst(fst),
+	m_wii(wii),
+	m_fst_size(fst_size),
+	m_index(index),
+	m_total_file_infos(total_file_infos)
+{ }
+
+// For the root object only
+CFileInfoGCWii::CFileInfoGCWii(const u8* fst, bool wii, u32 fst_size)
+	: m_fst(fst),
```

Comment:
This screams "delegated constructors!" to me. Could you use them to reduce duplication in your initializer lists?